xarray/core/dataset.py

```diff
@@ -1475,13 +1475,20 @@ def __bool__(self) -> bool:
     def __iter__(self) -> Iterator[Hashable]:
         return iter(self.data_vars)
 
-    def __array__(self, dtype=None):
-        raise TypeError(
-            "cannot directly convert an xarray.Dataset into a "
-            "numpy array. Instead, create an xarray.DataArray "
-            "first, either with indexing on the Dataset or by "
-            "invoking the `to_array()` method."
-        )
+    if TYPE_CHECKING:
+        # needed because __getattr__ is returning Any and otherwise
```

Comment:
This is the issue that is also causing this problem here: https://github.com/pydata/xarray/issues/4601